### PR TITLE
Add ARK 4IN1

### PR DESCRIPTION
**Pull request by AlexKlimaj**
https://arkelectron.com/product/ark-4in1-esc

IMG_3371-edited-Large

### CODE REVIEW

**Comment by freasy** [comment]
Please change the pr to draft, if it's not finished yet

**Comment by freasy** [comment]
Can you explain your ideas behind nfault etc?

Also please don't mix target prs and code prs. I have to ask you to separate those

File: Keil_Projects/Am32F051.uvoptx [hunk 1]
```diff
@@ -1148,7 +1148,7 @@
       <OPTFL>
         <tvExp>1</tvExp>
         <tvExpOptDlg>0</tvExpOptDlg>
-        <IsCurrentTarget>1</IsCurrentTarget>
+        <IsCurrentTarget>0</IsCurrentTarget>
```
Comment (tridge) [inline Keil_Projects/Am32F051.uvoptx:1151]: better to not have these keil changes in PRs

File: Mcu/f051/Src/peripherals.c [hunk 2]
```diff
@@ -495,6 +495,43 @@ void initLed()
 }
 #endif
 
+#ifdef USE_NSLEEP
+void initnSleep()
```
Comment (tridge) [inline Mcu/f051/Src/peripherals.c:499]: these are initialised but don't seem to be used? how do you envisage using them?

**Review by tridge — CHANGES_REQUESTED** [review]

File: Mcu/f051/Src/peripherals.c (hunk 2 above)
Comment (AlexKlimaj) [reply Mcu/f051/Src/peripherals.c:499]: Putting the gate drivers into sleep mode when idle saves about 600mW. Ideally when "disarmed" nSLEEP would be driven low, then when "armed" get driven high.

Comment (tridge) [reply Mcu/f051/Src/peripherals.c:499]: ok, sounds very worthwhile. Needs some comments and some hooks to use it. We'd also need to be careful with timing - we presumably need some delay between when we get a disarm signal and actually put the gate drivers to sleep.

Comment (AlexKlimaj) [reply Mcu/f051/Src/peripherals.c:499]: At the very least, I need this initializer here for the pin state.

**Comment by freasy** [comment]
Please separate target and code changes. The nSLEEP code should be an own PR, also always base from master and don't mix other PRs into this.

**Comment by AlexKlimaj** [comment]
I need the nSLEEP definition to set the initial pin state for the board
definition.

**Comment by AlexKlimaj** [comment]
Any update on this PR?

**Comment by AlexKlimaj** [comment]
Rebased on main.

**Review by AlkaMotors — APPROVED** [review]

**Review by tridge — APPROVED** [review]

**Pull request merged**